setup.py

```diff
@@ -101,11 +102,60 @@ def install_jupyter_hook(prefix=None, root=None):
             d, 'xonsh', user=user, replace=True, prefix=prefix)
 
 
+def dirty_version():
+    """
+    If install/sdist is run from a git directory (not a conda install), add
+    a devN suffix to reported version number and write a gitignored file
+    that holds the git hash of the current state of the repo to be queried
+    by ``xonfig``
+    """
+    try:
+        _version = subprocess.check_output(['git', 'describe', '--tags'])
+        _version = _version.decode('ascii')
```

Comment:
All the lines bellow this can be moved outside of the top-level try-except block.